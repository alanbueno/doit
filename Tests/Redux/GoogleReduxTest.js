import Actions, { reducer, INITIAL_STATE } from 'doit/App/Redux/GoogleRedux'
const token = '123'

test('request', () => {
  const state = reducer(INITIAL_STATE, Actions.listRequest(token))
  expect(state.fetching).toBe(true)
  expect(state.accessToken).toBe('123')
})

test('success', () => {
  const records = []
  const state = reducer(INITIAL_STATE, Actions.listSuccess(records))
  expect(state.fetching).toBe(false)
  expect(state.error).toBe(false)
  expect(state.listsAndTasks).toEqual([])
})

test('failure', () => {
  const state = reducer(INITIAL_STATE, Actions.listFailure())
  expect(state.fetching).toBe(false)
  expect(state.error).toBe(true)
})

test('setToken', () => {
  const state = reducer(INITIAL_STATE, Actions.setToken(token))
  expect(state.accessToken).toBe('123')
})

test('setFilter', () => {
  let filters = ['all', 'done', 'undone']
  filters.map(item => {
    const state = reducer(INITIAL_STATE, Actions.setFilter(item))
    expect(state.filters).toMatchSnapshot()
  })
})

test('updateTaskFailure', () => {
  const state = reducer(INITIAL_STATE, Actions.updateTaskFailure())
  expect(state.fetching).toBe(false)
  expect(state.error).toBe(true)
})
