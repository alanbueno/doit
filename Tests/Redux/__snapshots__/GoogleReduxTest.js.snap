// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setFilter 1`] = `
Array [
  Object {
    "active": true,
    "title": "all",
  },
  Object {
    "active": false,
    "title": "done",
  },
  Object {
    "active": false,
    "title": "undone",
  },
]
`;

exports[`setFilter 2`] = `
Array [
  Object {
    "active": false,
    "title": "all",
  },
  Object {
    "active": true,
    "title": "done",
  },
  Object {
    "active": false,
    "title": "undone",
  },
]
`;

exports[`setFilter 3`] = `
Array [
  Object {
    "active": false,
    "title": "all",
  },
  Object {
    "active": false,
    "title": "done",
  },
  Object {
    "active": true,
    "title": "undone",
  },
]
`;
